### DISCUSSION TITLE
Disable foreign key check in relations

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#2999] [Help]
I have a complicated relation query like below
```
@hasOne(() => ProductsPrice, {
    foreignKey: 'productsId',
    onQuery: (builder) => {
      builder
        .joinRaw(
          `LEFT JOIN products_offers as po ON po.products_id = products_prices.products_id 
            AND po.is_published > 0 
            AND NOW() BETWEEN started_at AND ended_at 
            AND (po.stock_count >= 1 OR po.stock_count is null)`
        )
        .select(
          Database.raw(
            `:column1 as loyalty_level, 
              products_prices.products_id as products_id,
              CASE po.type
                WHEN "FIXED" THEN po.amount
                WHEN "MINUS" THEN products_prices.price - po.amount
                else products_prices.price
              end as offer,
              products_prices.bronze_price as bronze,
              products_prices.silver_price as silver,
              products_prices.gold_price as gold,
              GREATEST(IFNULL(products_prices.market_price, 0), products_prices.price) as base,
                case 
                    WHEN po.amount IS NULL THEN "regular"
                    else "special"
                end AS offer_type`,
            {
              column1: null,
            }
          )
        )
    },
```

in the executaion time I got `ambiguous` error about `productsId` column
for resolve my error I should set foreign key to be `products_prices.products_id`
and If I do this I got the  `E_MISSING_MODEL_ATTRIBUTE` error on execution time

How should I resolve this problem?

**Comment by McSneaky** [comment]
I guess error comes from this line: `products_prices.products_id as products_id`
You can try to remove it totally, since nothing seems to be using products_id (at least not in that query)
And assuming it's the same as on model itself it shouldn't need to exist. It will just collide with model one

If you still need that you can give it some other alias rather than `products_id`, `prod_id` for example

**Reply by farshadfahimi** [reply]
I check the query in many ways, at list when I'm `load` the relation or use `eager loading` I got error on `foreignKey: 'productsId'` because thr query builder or eager loading append this to the query.

and after lot's of check in some others way, I defined this problem could be happend in some other types.
for example you can check the issue created by me on the [lucid soft delete package](https://github.com/radmen/adonis-lucid-soft-deletes/issues/26)

all the same problem could be solve if the query builder create the query with using the table name like `table`.`field`.

and I suggest this many time, I like to know why you don't use this syntax?
